### ISSUE TITLE
Autocomplete + Syntax error highlighting

### ISSUE THREAD

**Issue by Tom-III** [LADSoft/OrangeC#191]
Would it be possible to have tab-autocomplete and syntax error highlighting?  It would make development much smoother with the IDE.

Once again thank you for designing such a great compiler!  Amazing how there is DOS support and works on Windows 95/98 and doesn't introduce artificial incompatibility like MSVC does with its CRT.

**Comment by LADSoft** [comment]
Actually it is supposed to be doing autocomplete, for structure members and for function definitions.   There is a small program 'occpr' which is basically the C compiler accepting connections from the IDE over a pipe, and which writes to an SQL database.   That is one of several databases OCIDE reads.

Occpr is the same program that feeds in information about symbols so the ide knows how to colorize things...

I was wondering what was up with the code completion not working the other day but there is a lot going on so I haven't had a chance to look into it.   I will see what it is doing this weekend.

As far as syntax highlighting, I don't know if the present scheme would work without some modification.   The occpr approach to retrieiving information about the code is kind of slow...   and it would have to be modified to give feedback about errors.   I was wondering what would happen if I went to some effort to pull out various C/C++ operations that just don't make sense in this context but haven't gotten there yet...

**Comment by GitMensch** [comment]
In general I'd suggest to add a "syntax check only" option to `occ` (also: never invokes the linker) - a mode `occpr` would default to.

**Comment by Tom-III** [comment]
It doesn't have to have advanced stuff like parameter type checking, just the basics like mismatched parenthesis, forgotten semicolons, mismatched brackets, curly brackets would be a huge improvement.

**Comment by LADSoft** [comment]
that doesn't sound too hard...  worst of it will be drawing a little red squiggle!   Meanwhile I've improved the code completion, if you don't feel like snagging it from the CI I'll have a release in a week or so whenever I get the syntax highlighting worked out.

**Comment by LADSoft** [comment]
well it is not perfect but I'm open to suggestions...   will try to have a release within a few days.

**Comment by Tom-III** [comment]
Autocomplete still doesn't show up?
![Autocomplete not showing up](https://i.imgur.com/V32cmKh.png)

**Comment by chuggafan** [comment]
Is that a CI build or is that a release bui;d? because those are two different beasts

**Comment by Tom-III** [comment]
I got it from here https://ci.appveyor.com/project/LADSoft/orangec/build/job/p2eyjvjdewgk7i8j/artifacts

**Comment by LADSoft** [comment]
yeah that is the CI build.   I found out today that the database stuff isn't working in the CI builds...   checking into that is next on my list :)   I'm also cleaning up the syntax checking a little more.

**Comment by GitMensch** [comment]
@LADSoft Do you consider this close-fixed already?

**Comment by LADSoft** [comment]
I was intentionally leaving it open because I didn't know if there would be issues.   I may close it since now you've opened #228...  let me think about it...

**Comment by JakoDel** [comment]
Hi @LADSoft, is there any update on this matter? and thanks a lot for your work btw!

**Comment by LADSoft** [comment]
I haven't been working on the IDE; syntax highlighting probably doesn't work so well...
